src/adapters/Database.re

```diff
@@ -276,6 +279,47 @@ let hasUnfinishedHelpItem = userId =>
         | `Mutation(mutation) => Js.log2("MUTATION: ", mutation)
         };
 
+        MySql2.Connection.close(conn);
+      },
+    );
+  });
+
+let getQueuePosition = userId =>
+  Js.Promise.make((~resolve, ~reject as _reject) => {
+    Js.log2("USER_ID: ", userId);
+    let conn = connect();
+
+    let params =
+      MySql2.Params.named(
+        Json.Encode.(object_([("user_id", string(userId))])),
+      );
+
+    MySql2.execute(
+      conn,
+      "SELECT `id`,
+      (SELECT COUNT(*) FROM `help_items` WHERE `user_id` <= :user_id) AS `position`,
+      `user_id`,
+      `description`,
+      `room`,
+      `finished`,
+      `time_created`,
+      `time_closed`
+      FROM `help_items`
+      WHERE `user_id` = :user_id AND finished = false LIMIT 1;",
+      Some(params),
+      res => {
+        switch (res) {
+        | `Error(e) => Js.log2("ERROR: ", e)
+        | `Select(select) =>
+          Js.log2("Select: ", select);
+          let rows =
+            select
+            ->MySql2.Select.rows
+            ->Belt.Array.map(item => item |> Decode.helpItem);
```

Comment:
As discussed earlier I think we should skip `position` in the helpItem since it's dynamic and there's no point of holding on to it when a ticket is closed. It'll only ever be accurate for the response when someone requests mayday unless we keep updating it! 🙂 

I propose we calculate when called either by updating the SQL-query or by doing something like:

```sql
SELECT * from help_items WHERE finished = false ORDER BY time_created ASC LIMIT 1
```

... and then get the index from something like an array-method or similar.

Reply:
@lessp The position field isn't actually available in the table row, it gets calculated in the query and gets represented as "position" in a pseudo column.

But I will try to implement a better solution so that we don't need it in the type definition

Reply:
Ahh, I see!

Reply:
Well, if that’s case couldn’t you just skip decoding it as a helpItem and just get the position as an int and return that?

Reply:
Good point. I will look into doing that tomorrow afternoon